extractor/src/main/java/org/schabi/newpipe/extractor/utils/StringUtils.java

```diff
@@ -46,4 +45,47 @@ public static String matchToClosingParenthesis(@Nonnull final String string,
 
         return string.substring(startIndex, endIndex);
     }
+
+    private static int findNextParenthesis(@Nonnull final String string,
+                                           final int offset,
+                                           final boolean onlyOpen) {
+        boolean lastEscaped = false;
+        char quote = ' ';
+
+        for (int i = offset; i < string.length(); i++) {
+            boolean thisEscaped = false;
+            final char c = string.charAt(i);
+
+            switch (c) {
+                case '{':
+                    if (quote == ' ') {
+                        return i;
+                    }
+                    break;
+                case '}':
+                    if (!onlyOpen && quote == ' ') {
+                        return i;
+                    }
+                    break;
+                case '\\':
+                    if (!lastEscaped) {
+                        thisEscaped = true;
+                    }
+                    break;
+                case '\'':
+                case '"':
```

Comment:
The grave character (`) should also probably be included since they can be used to create multi-line strings.